### PR TITLE
[MM-26589] Add context version to all public methods

### PR DESCRIPTION
**Pull request by ethervoid**
#### Summary

To be able to cancel the request by timeout or deadline we need to pass the context to the request.
    
This PR a new *WithContext method to all the exposed methods so the user can:

- Use the current method, which is going to create a [context.Background](https://golang.org/pkg/context/#Background) context
- Use the *WithContext method and pass their own context

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-26589

### CODE REVIEW

**Comment by ethervoid** [comment]
@agnivade given that by now I don't have permission to assign reviews I summon you!

File: circleci_test.go [hunk 1]
```diff
@@ -101,12 +103,28 @@ func TestClient_request(t *testing.T) {
...
+	defer teardown()
+	client.Token = "ABCD"
+	mux.HandleFunc("/me", func(w http.ResponseWriter, r *http.Request) {
+		time.Sleep(1 * time.Second)
+	})
+
+	ctx, cancel := context.WithTimeout(context.Background(), 0*time.Second)
+	defer cancel()
+	err := client.request(ctx, "GET", "/me", &User{}, nil, nil)
+	if !strings.Contains(err.Error(), "context deadline exceeded") {
```
Comment (agnivade) [inline circleci_test.go]: We can use `errors.As(err, &context.DeadlineExceeded)`

File: circleci_test.go [hunk 2]
```diff
@@ -968,6 +986,24 @@ func TestClient_GetActionOutput_noOutput(t *testing.T) {
...
+		testMethod(t, r, "GET")
+		fmt.Fprintf(w, `[{"Message":"hello"}, {"Message": "world"}]`)
+	})
+
+	action := &Action{HasOutput: true, OutputURL: server.URL + "/some-s3-path"}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Microsecond)
+	defer cancel()
+	_, err := client.GetActionOutputsWithContext(ctx, action)
+	if err == nil || !strings.Contains(err.Error(), "context deadline exceeded") {
```
Comment (agnivade) [inline circleci_test.go]: Same as above.

**Review by agnivade — CHANGES_REQUESTED** [review]
Just a small thing. Looks good otherwise.

**Comment by ethervoid** [comment]
Don't know how this was closed :scream:

**Pull request closed without merging**